src/playlist-loader/dash-main-playlist-loader.js

```diff
@@ -0,0 +1,181 @@
+import PlaylistLoader from './playlist-loader.js';
+import {resolveUrl} from '../resolve-url';
+import {parse as parseMpd, parseUTCTiming} from 'mpd-parser';
+import {mergeManifest, forEachPlaylist} from './utils.js';
+
+/**
+ * An instance of the `DashMainPlaylistLoader` class is created when VHS is passed a DASH
+ * manifest. For dash main playlists are the only thing that needs to be refreshed. This
+ * is important to note as a lot of the `DashMediaPlaylistLoader` logic looks to
+ * `DashMainPlaylistLoader` for guidance.
+ *
+ * @extends PlaylistLoader
+ */
+class DashMainPlaylistLoader extends PlaylistLoader {
```

Comment:
This class deals with getting the sever clock offset and merging the old main manifest into the new one.